app/bundles/PageBundle/Entity/HitRepository.php

```diff
@@ -147,6 +148,11 @@ public function getHitCountForSource($source, $sourceId = null, $fromDate = null
      */
     public function getEmailClickthroughHitCount($emailIds, \DateTime $fromDate = null, $code = 200)
     {
+        $cacheStorageHelper = new CacheStorageHelper(CacheStorageHelper::ADAPTOR_FILESYSTEM);
```

Comment:
CacheStorageHelper is deprecated in favor of CacheBundle https://developer.mautic.org/#cache-bundle. You should use service mautic.cache.provider

I think repository does not allow load services in construct, but you can use methodCall like here https://github.com/mautic/mautic/blob/93813f25837ec83783078193f9b6a7fb68f714fb/app/bundles/LeadBundle/Config/config.php#L914

All move logic to model or service.

Reply:
Hi @kuzmany , i moved logic of cache into EmailModel, it already used CachedStorageHelper so i used it to cache the hits.
I hope it goes well now.